pkg/authorization/registry/rolebinding/policybased/virtual_storage.go

```diff
@@ -148,7 +148,9 @@ func (m *VirtualStorage) createRoleBinding(ctx apirequest.Context, obj runtime.O
 
 	// Retry if we hit a conflict on the underlying PolicyBinding object
 	if err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
-		policyBinding, err := m.getPolicyBindingForPolicy(ctx, roleBinding.RoleRef.Namespace, allowEscalation)
+		// get or auto create policy binding so we can deprecate policy and policy binding objects in 3.6
+		// thus normal users can always create a role binding referring to a role
+		policyBinding, err := m.getPolicyBindingForPolicy(ctx, roleBinding.RoleRef.Namespace, true)
```

Comment:
This would allows someone to create a rolebinding pointing to a role in another namespace.  That is an anti-goal.